examples/react-hooks-example/src/App.js

```diff
@@ -0,0 +1,134 @@
+import React, {useEffect, useState} from 'react';
+import {
+    Box,
+    Grommet,
+    DataTable,
+    Text
+} from 'grommet';
+import { pipe } from 'rxjs';
+import Subspace, {$latest} from '@embarklabs/subspace';
+import Web3 from 'web3';
+import exchangeABI from './contract/exchange_abi.json'
+
+const web3 = new Web3('https://cloudflare-eth.com');
+const subspace = new Subspace(web3.currentProvider);
+var dai = new web3.eth.Contract(exchangeABI, '0x2a1530C4C41db0B0b2bB646CB5Eb1A67b7158667');
+const daiContract = subspace.contract(dai);
+subspace.init();
```

Comment:
Consider updating to Subspace 2.0.

Reply:
Happy to update this one.  You're just referring to passing the whole `Web3` object in the subspace constructor rather than the provider?

Reply:
Yes, as well as adding an `await` to `subspace.init()` to use Subspace only after it has initialized

Reply:
Done.  Switched to the `SubspaceProvider` and `useSubspace` components from `subspace-react`.